### PR TITLE
build: Update to C# 11

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -48,7 +48,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <LangVersion>10.0</LangVersion>
+    <LangVersion>latest</LangVersion>
```
Comment (jeromelaban) [inline src/Directory.Build.props:51]: Is 11.0 not supported ? (I'd rather have it be explicit so we don't fall back on what the current targetframework supports).

Comment (Youssef1313) [reply src/Directory.Build.props:51]: @jeromelaban It's supported but not for all projects it seems.

Comment (Youssef1313) [reply src/Directory.Build.props:51]: I'll see if I could use 11 here and adjust the failing projects.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**